napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -164,81 +250,184 @@ def setup_ui(self, enabled=True):
+        self.info_choice_wdg.setObjectName("install_choice_widget")
+        self.info_choice_wdg.hide()
+
+        self.cancel_btn = QPushButton("Cancel", self)
+        self.cancel_btn.setSizePolicy(sizePolicy)
+        self.cancel_btn.setObjectName("remove_button")
+        self.row2.addWidget(
+            self.cancel_btn, 0, 8, 1, 1, alignment=Qt.AlignmentFlag.AlignTop
+        )
+
+        self.action_button = QPushButton(self)
+        self.action_button.setFixedWidth(70)
+        sizePolicy1 = QSizePolicy(QSizePolicy.Preferred, QSizePolicy.Preferred)
+        self.action_button.setSizePolicy(sizePolicy1)
+        self.row2.addWidget(
+            self.action_button, 0, 8, 1, 1, alignment=Qt.AlignmentFlag.AlignTop
+        )
+
         self.v_lay.addLayout(self.row2)
 
+        self.info_widget = QWidget(self)
+        self.info_widget.setLayoutDirection(Qt.LeftToRight)
+        self.info_widget.setObjectName("info_widget")
+        info_layout = QGridLayout()
+        info_layout.setContentsMargins(0, 0, 0, 0)
+        info_layout.setVerticalSpacing(0)
+        self.version_text = QLabel('Version:')
+        self.package_name = QLabel()
+        self.source_text = QLabel('Source:')
+        self.source = QLabel('PyPI')
+
+        info_layout.addWidget(self.source_text, 0, 0)
+        info_layout.addWidget(self.source, 1, 0)
+        info_layout.addWidget(self.version_text, 0, 1)
+        info_layout.addWidget(self.package_name, 1, 1)
+
+        self.install_info_button.setFixedWidth(150)
+        self.install_info_button.layout().setContentsMargins(0, 0, 0, 0)
+        self.info_widget.setLayout(info_layout)
+
+    def _populate_version_dropdown(self, source: Literal["PyPI", "Conda"]):
+        """Display the versions available after selecting a source: pypi or conda."""
+        if source == 'PyPI':
+            versions = self._versions_pypi
+        else:
+            versions = self._versions_conda
+        self.version_choice_dropdown.clear()
+        for version in versions:
+            self.version_choice_dropdown.addItem(version)
+
     def _on_enabled_checkbox(self, state: int):
         """Called with `state` when checkbox is clicked."""
         enabled = bool(state)
         plugin_name = self.plugin_name.text()
-        pm2 = PluginManager.instance()
+        pm2 = npe2.PluginManager.instance()
         if plugin_name in pm2:
             pm2.enable(plugin_name) if state else pm2.disable(plugin_name)
             return
 
-        for npe1_name, _, distname in plugin_manager.iter_available():
+        for (
+            npe1_name,
+            _,
+            distname,
+        ) in napari.plugins.plugin_manager.iter_available():
             if distname and (normalized_name(distname) == plugin_name):
-                plugin_manager.set_blocked(npe1_name, not enabled)
+                napari.plugins.plugin_manager.set_blocked(
+                    npe1_name, not enabled
+                )
```

Comment:
Is it possible to be more than one entry satisfying the condition?

Reply:
I don't think so?   Can you think of a place where this would have more than 1 plugin installed that matches the plugin name on the dialog?

Reply:
No, this code just does not stop after the first match but continues iterations.

Reply:
Oh, I see what you mean.  yes, will change this.